File: actividades/prog/ejemplos/holamundos/j-canarias/src/aplicaciones/terminal/ui/UITextoPlano.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package terminal.ui;

import terminal.util.ConsolaIO;
import terminal.shell.InterfazShell;

/**
 *
 * @author david
 */
public class UITextoPlano implements InterfazUI {

    public void iniciar(InterfazShell con) {

        ConsolaIO.println("\nEntrando en consola virtual...");
        ConsolaIO.printlnSimbolo('<', 35);

        ConsolaIO.print(con.getSalida());
        
        while (!con.isFin()) {
            con.setEntrada(ConsolaIO.readLine());
            ConsolaIO.print(con.getSalida());
        }
        ConsolaIO.printlnSimbolo('>', 35);
        ConsolaIO.println("Consola virtual cerrada.\n");
    }
}

File: actividades/prog/files/ruby/mil-ejemplos/aplicaciones/terminal/clases/ui/ui_texto_plano.rb
# 
# To change this template, choose Tools | Templates
# and open the template in the editor.
 

class UITextoPlano
    
  def consola=(n)
    @shell = n
  end
  
  def iniciar
    print @shell.prompt
    while !@shell.fin?
      orden = gets.chomp!
      @shell.ejecutar orden
      puts @shell.salida
      print @shell.prompt
    end
  end
end
